File: src/main/java/org/bipmed/brave/server/variant/Variant.java
package org.bipmed.brave.server.variant;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document
@Data
@Builder
public class Variant {

    @JsonIgnore
    @Id
    private String id;
    private List<String> snpIds;
    private String datasetId;
    private String assemblyId;
    private Integer totalSamples;
    private String referenceName;
    private Long start;
    private String referenceBases;
    private List<String> alternateBases;
    private List<String> geneSymbol;
    private List<Float> alleleFrequency;
    private Long sampleCount;

    private Statistics coverage;
    private Statistics genotypeQuality;

    private String clnsig;
    private List<String> hgvs;
    private List<String> type;

    @Data
    class Statistics {

        private Integer min;
        private Float q25;
        private Float median;
        private Float q75;
        private Integer max;
        private Float mean;
    }
}

File: src/main/kotlin/org/bipmed/brave/server/variant/Variant.kt
package org.bipmed.brave.server.variant

import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Variant(
        @JsonIgnore
        @Id
        val id: String? = null,
        val snpIds: List<String> = emptyList(),
        val datasetId: String,
        val assemblyId: String,
        val totalSamples: Int,
        val referenceName: String,
        val start: Long,
        val referenceBases: String,
        val alternateBases: List<String>,
        val geneSymbol: List<String>? = null,
        val alleleFrequency: List<Number> = emptyList(),
        val sampleCount: Long? = null,

        val coverage: Statistics? = null,
        val genotypeQuality: Statistics? = null,

        val clnsig: String? = null,
        val hgvs: List<String>? = null,
        val type: List<String>? = null
) {
    data class Statistics(
            val min: Int? = null,
            val q25: Number? = null,
            val median: Number? = null,
            val q75: Number? = null,
            val max: Int? = null,
            val mean: Number? = null
    )
}